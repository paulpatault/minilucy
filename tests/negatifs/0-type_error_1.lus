(* x when 1 mal typé, 1 devrait être de type bool *)

node main0 (x: int) returns (o: bool);
let
  o = x + (x when 1);
tel
