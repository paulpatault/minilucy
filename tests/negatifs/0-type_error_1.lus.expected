
/**************************************/
/* Sugar automaton print              */
/**************************************/
type inductive_bool =
  | False
  | True

node main0(x : int) returns (o : bool);
var ;
let
  (o) = add(x, x when True(1));
tel

[31m[1mTyping error: [0mthis expression has type int but is expected to have type bool

